pkg/azurefile/controllerserver.go

```diff
@@ -1205,3 +1232,28 @@ func isValidVolumeCapabilities(volCaps []*csi.VolumeCapability) error {
 	}
 	return nil
 }
+
+func generateSASToken(accountName, accountKey, storageEndpointSuffix string) (string, error) {
+	credential, err := service.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	serviceClient, err := service.NewClientWithSharedKeyCredential(fmt.Sprintf("https://%s.file.%s/", accountName, storageEndpointSuffix), credential, nil)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new client with shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	nowTime := time.Now()
+	sasURL, err := serviceClient.GetSASURL(
+		sas.AccountResourceTypes{Object: true, Service: true, Container: true},
```

Comment:
since this is azure file share copy, set `Object` and `Container` as false?

Reply:
done, just set service ture.

Reply:
azcopy also need `object` account resource type permission.

Reply:
`Container` should also needed, when it set false it will failed when create vhd file in sanity test and e2e test